src/main/java/edu/whimc/portals/Portal.java

```diff
@@ -23,6 +24,10 @@
 import org.bukkit.permissions.Permission;
 import org.bukkit.util.Vector;
 
+/**
+ * A teleportation tool for players to move in game and also
+ * a managing class for all active portals and their behavior.
+ */
 public class Portal {
 
     private static List<Portal> portals = new ArrayList<>();
```

Comment:
@jackah2 The Portal file gets a little too busy with these static fields and methods. While the behavior of these cached portals and destinations are basically static, I am of the mind that storage should be handled by storage classes, separate from any base object class to reduce confusion and clutter. I like what you did with the LocationSaver; maybe another portal/destination handler could be added as a field in Main and then all command classes and any future APIs can use Main's instance of the handler instead of using a bunch of static fields bundled into the Portal class. 

It also doesn't really matter because I don't think there are plans to expand on this plugin so reducing clutter might not be important anyway. This is also my organizational opinion and maybe static fields are clearer.

Reply:
I definitely agree that there is a need for abstraction here. In some of the other plugins, I use a manager with static fields to take care of creating, removing, and fetching portals.